### DISCUSSION TITLE
security issue

### DISCUSSION THREAD

**Discussion by soltanianmj** [supabase/supabase#16609] [Questions] [answered]
i develop an app with Supabase and NextJs and store api key and anon key in a .env file but on network tab you can see my api key and anon key. is it a security issue or is fine?

**Comment by GaryAustin1** [comment, marked as answer]
This is the normal way to use Supabase on a browser.  You cannot hide any headers or parameters from the browser dev.  This is also why you need to use RLS to limit operations that can be done.  Never use your service_role key in a browser environment or where it is exposed to the public though.
There are many discussions searching here or the web.
https://github.com/orgs/supabase/discussions/256

**Reply by soltanianmj** [reply]
so whats the use of storing anon key in .env file when someone with some role can access it?

**Reply by GaryAustin1** [reply]
NEXT_PUBLIC_SUPABASE_ANON_KEY, if that is what you are using, means it is available and can be used on browsers is my understanding (I don't use Nextjs).   Anything on a browser can be seen by the user using the developer console.   You could also have it as a const in your code if you desired to do it that way.   I do for my pure javascript test sites.

**Reply by soltanianmj** [reply]
that's true the name describes it.'NEXT_PUBLIC_SUPABASE_ANON_KEY'
thanks for your time
